### PR TITLE
feat: parse arguments for commands in webpack-cli

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
feature + refactor

**Did you add tests for your changes?**
Updated tests for to match it

**If relevant, did you update the documentation?**
No, but need to update for migrate command

**Summary**
Avoid using new instance of `webpack-cli` for parsing arguments in commands.

**Does this PR introduce a breaking change?**
No

**Other information**
/cc @alexander-akait

### CODE REVIEW

**Review by alexander-akait — COMMENTED** [review]
Looks good for refactor, ideally we should pass `parser` in sum command and inside sub command register new command, interface:
```js
class MyCommand {
  apply(parser) {
    // Logic for registration command and flags for command
  }
}
```

**Comment by alexander-akait** [comment]
@rishabh3112 Can you fix CI?

**Comment by rishabh3112** [comment]
Yup on it

File: packages/info/__tests__/info.test.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import info from '../src/index';
 
 describe('info tests', () => {
     it('should log environment info', async () => {
-        const envInfo = await info();
+        const envInfo = await info({});
```
Comment (alexander-akait) [inline packages/info/__tests__/info.test.ts:5]: Why we need `{}`, can you do default `{}` if nothing passed

Comment (rishabh3112) [reply packages/info/__tests__/info.test.ts:5]: Okay, will do

Comment (rishabh3112) [reply packages/info/__tests__/info.test.ts]: done

**Comment by alexander-akait** [comment]
WIP on this, need more improvements

**Comment by alexander-akait** [comment]
Fixed in master, anyway thanks for the PR

**Pull request closed without merging**